controlplane/kubeadm/controllers/scale.go

```diff
@@ -135,5 +135,9 @@ func selectMachineForScaleDown(controlPlane *internal.ControlPlane, outdatedMach
 	if outdatedMachines.Len() > 0 {
 		machines = outdatedMachines
 	}
+	withDeleteAnnotation, err := controlPlane.MachineWithDeleteAnnotation(machines)
+	if err == nil {
+		return withDeleteAnnotation, nil
+	}
```

Comment:
```suggestion
	if withDeleteAnnotation, err := controlPlane.MachineWithDeleteAnnotation(machines); err == nil {
		return withDeleteAnnotation, nil
	}
```
Also given that error is always nil, should this check if there is at least one  machine withDeleteAnnotation?